google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -515,70 +756,117 @@ private void verifyNotClosed() {
     }
   }
 
+  /**
+   * Attaches a listener that is run every time a BulkWriter operation successfully completes.
+   *
+   * <p>For example, see the sample code: <code>
+   *   BulkWriter bulkWriter = firestore.bulkWriter();
+   *   bulkWriter.addWriteResultListener(
+   *         new WriteResultCallback() {
+   *           public void onResult(DocumentReference documentReference, WriteResult result) {
+   *             System.out.println(
+   *                 "Successfully executed write on document: "
+   *                     + documentReference
+   *                     + " at: "
+   *                     + result.getUpdateTime());
+   *           }
+   *         });
+   * </code>
+   *
+   * @param writeResultCallback A callback to be called every time a BulkWriter operation
+   *     successfully completes.
+   */
+  public void addWriteResultListener(WriteResultCallback writeResultCallback) {
+    successListener = writeResultCallback;
+  }
+
+  /**
+   * Attaches an error handler listener that is run every time a BulkWriter operation fails.
+   *
+   * <p>BulkWriter has a default error handler that retries UNAVAILABLE and ABORTED errors up to a
+   * maximum of 10 failed attempts. When an error handler is specified, the default error handler
+   * will be overwritten.
+   *
+   * <p>For example, see the sample code: <code>
+   *   BulkWriter bulkWriter = firestore.bulkWriter();
+   *   bulkWriter.addWriteErrorListener(
+   *         new WriteErrorCallback() {
+   *           if (error.getStatus() == Status.UNAVAILABLE
+   *                 && error.getFailedAttempts() < MAX_RETRY_ATTEMPTS) {
+   *               return true;
+   *             } else {
+   *               System.out.println("Failed write at document: " + error.getDocumentReference());
+   *               return false;
+   *             }
+   *         });
```

Comment:
You should probably use Lambda syntax here. If you stick with the current example, you need to add the function signature.

Reply:
done.